lms/static/js/learner_dashboard/views/collection_list_view.js

```diff
@@ -1,22 +1,41 @@
-;(function (define) {
+;
+(function (define) {
     'use strict';
 
-    define(['backbone'],
-         function(
-             Backbone
-         ) {
+    define(['backbone',
+            'jquery',
+            'underscore',
+            'gettext',
+            'text!../../../templates/learner_dashboard/empty_programs_list.underscore'
+        ],
+        function (Backbone,
+                  $,
+                  _,
+                  gettext,
+                  emptyProgramsListTpl) {
             return Backbone.View.extend({
-                initialize: function(data) {
+                initialize: function (data) {
                     this.childView = data.childView;
+                    this.context = data.context;
                 },
 
-                render: function() {
-                    var childList = [];
-                    this.collection.each(function(program){
-                        var child = new this.childView({model:program});
-                        childList.push(child.el);
-                    }, this);
-                    this.$el.html(childList);
+                render: function () {
+                    var childList, tpl;
+                    if (!this.collection.length) {
```

Comment:
Nitpick: there should be a new line between the var declaration and the if block.